packages/firestore/src/core/sync_engine.ts

```diff
@@ -680,12 +676,38 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     delete this.queryViewsByTarget[queryView.targetId];
 
     if (this.isPrimary) {
+      const limboKeys = this.limboDocumentRefs.referencesForId(
+        queryView.targetId
+      );
       this.limboDocumentRefs.removeReferencesForId(queryView.targetId);
-      await this.gcLimboDocuments();
+      limboKeys.forEach(async limboKey => {
```

Comment:
I have to run to a meeting, but in case I forget this later: I don't think this works. You have to append this Promise to another data structure and then wait on all those promises. ForEach doesn't wait on the Promise resolution.

Reply:
I suspect you're right. I've updated it with a chain of `PersistencePromise` with a final `await p.toPromise()` at the end.